src/command/repo/get.c

```diff
@@ -49,10 +68,78 @@ storageGetProcess(IoWrite *destination)
                 // Check for a passphrase parameter
                 const String *cipherPass = cfgOptionStrNull(cfgOptCipherPass);
 
-                // If not passed as a parameter use the repo passphrase
+                // If not passed as a parameter then determine the passphrase using the following pattern:
+                //
+                // REPO / (repo passphrase)
+                //      / archive / (repo passphrase)
+                //      / archive / stanza / (archive passphrase)
+                //      / backup  / (repo passphrase)
+                //      / backup  / stanza / (backup passphrase)
+                //      / backup  / stanza / set / (manifest passphrase)
+                //      / backup  / stanza / backup.history / (backup passphrase)
+                // -----------------------------------------------------------------------------------------------------------------
                 if (cipherPass == NULL)
+                {
                     cipherPass = cfgOptionStr(cfgOptRepoCipherPass);
 
+                    // Encrypted files could be stored directly at the root of the repo if needed
+                    // We should then at least be able to determine the archive or backup directory
+                    StringList *filePathSplitLst = strLstNewSplit(file, FSLASH_STR);
+
+                    if (strLstSize(filePathSplitLst) > 1)
```

Comment:
I feel like this should be `strLstSize(filePathSplitLst) > 1`. The minimum path that makes sense is archive|backup/<stanza>/<file>, right?

Reply:
Isn't it already the case ? Or do you mean > 2 ?

After the repo-path is removed from the file path, indeed, 1st argument should be archive or backup. The next one should be the stanza name. If we look at what are inside the stanza directories, it should be `strLstSize(filePathSplitLst) > 2` then...

Reply:
Whoops! Yes, that's what I meant.

Reply:
Ok, I'll update it, thanks

Reply:
Excellent. This pretty much looks ready to me once this is done.

Reply:
I just have to figure out where the error comes from in the mock tests:

```
2020-07-16 09:13:07.016 P00  ERROR: [032]: /home/travis/test/bin/none/pgbackrest --config=/home/travis/test/test-0/db-primary/pgbackrest.conf --stanza=db --log-level-console=off --log-level-stderr=error --no-repo1-azure-verify-tls --repo1-azure-host=172.17.0.2  --ignore-missing repo-get /backup/db/backup.info terminated unexpectedly [032]: ERROR: [032]: unable to determine encryption key for 'ackup/db/backup.info'
```

It looks like the repo-path would only be '/'. I'll have a look at how the `file = strSub(file, strSize(cfgOptionStr(cfgOptRepoPath)) + 1);` reacts when repo-path ends with/without /

Reply:
In other places we handle this as a special case and that is probably needed here as well.

Reply:
Yeah, that looks right to me. I'll have a closer look later and commit if everything looks OK.

Reply:
I think there are still issues with MockStanzaTest. It tries to `repo-get /archive/db/archive.info.old`. 
We defined only `.info` and `.info.copy` are authorized to use the repoCipherPass. Otherwise, it will try to get the archiveCipherPass.
Should we allow files on the same directory level as `.info` to use the repoCipherPass ? Or should we adjust the tests ?

Reply:
I actually have a PR in to get rid of these tests, but for now I think it would be OK to copy this file raw. Let me have a look at it and see where it would be best to adjust the integration test.

Reply:
Great, thanks!

Reply:
Just pushed a commit up that will hopefully fix it. This was arguably a bug in the integration tests which was just exposed by the stricter rules in repo-get.

Reply:
Well, that was a good catch, it fixed the tests :-)